src/ProfileRight.php

```diff
@@ -302,38 +302,43 @@ public static function updateProfileRights($profiles_id, array $rights = [])
     public function post_addItem($history = true)
     {
         // Refresh session rights to avoid log out and login when rights change
-        $this->forceCurrentSessionRights($this->fields['profiles_id'], $this->fields['name'], $this->fields['rights']);
+        $this->updateProfileLastRightsUpdate($this->fields['profiles_id']);
     }
 
     public function post_updateItem($history = true)
     {
         // Refresh session rights to avoid log out and login when rights change
-        $this->forceCurrentSessionRights($this->fields['profiles_id'], $this->fields['name'], $this->fields['rights']);
+        $this->updateProfileLastRightsUpdate($this->fields['profiles_id']);
     }
 
     /**
-     * Force rights for given rightname on current session.
+     * Update last rights update for given profile.
      *
      * @param int $profile_id
-     * @param string $rightname
-     * @param int $rights
      * @return void
      */
-    private function forceCurrentSessionRights(int $profile_id, string $rightname, int $rights): void
+    private function updateProfileLastRightsUpdate(int $profile_id): void
     {
-        if (
-            isset($_SESSION['glpiactiveprofile']['id'])
-            && $_SESSION['glpiactiveprofile']['id'] == $profile_id
-            && (
-                !isset($_SESSION['glpiactiveprofile'][$rightname])
-                || $_SESSION['glpiactiveprofile'][$rightname] != $rights
-            )
-        ) {
-            $_SESSION['glpiactiveprofile'][$rightname] = $rights;
-            unset($_SESSION['glpimenu']);
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        // Check if the 'last_rights_update' field exists before trying to update it.
+        // This field may not exist yet as it is added by a migration, and other migrations
+        // that add a right could be executed before the migration that adds this field.
+        if (!$DB->fieldExists('glpi_profiles', 'last_rights_update')) {
+            return;
         }
-    }
 
+        $DB->updateOrDie(
```

Comment:
You should use the `Profile::update()` method. Indeed, plugins may want add a hook on that change, for instance to invalidate some tokens they generated.

It will not necessary to check `$DB->fieldExists('glpi_profiles', 'last_rights_update')` with this method.